### ISSUE TITLE
Design of organization detail page 

### ISSUE THREAD

**Issue by manjitapandey** [hotosm/fmtm#1572]
**Is your feature request related to a problem? Please describe.**
Currently, Users are not able to view the details of organisation created and no way to navigate to organisation editing page. 

**Describe the solution you'd like**
An organisation details page for each organisation. 
![image](https://github.com/hotosm/fmtm/assets/97273021/d4b5bcbf-f574-43e2-afae-4629ddbdead7)

https://www.figma.com/design/GUKooEhrNwHFTaAaEHxUcS/FMTM?node-id=2332-2902&t=odBrqjBb2gOY1UXD-0

**Describe alternatives you've considered**
We can have further discussion in comments whether this page should be accessible only to organisation admins or other users too.

**Comment by manjitapandey** [comment]
The latest design og organization details and editing pages are ready. Please provide your feedback on design. 

![Image](https://github.com/user-attachments/assets/397671fc-815d-45de-995e-63eb77a49ba5)

![Image](https://github.com/user-attachments/assets/8cc23a26-081b-4635-934a-abc344e6bbd5)
OR
![Image](https://github.com/user-attachments/assets/79e4517d-3d12-43ed-b475-97c2e692decd)

**Comment by spwoodcock** [comment]
I see TM had a public organisation page, so it probably makes sense being public here

![Screenshot_20250127-081646.png](https://github.com/user-attachments/assets/3123516b-acce-4076-9c80-79636cebb0aa)

Seeing this, it makes complete sense to me building this as a reusable component across TM, FieldTM, DroneTM (to avoid duplication). Will give that some thought.

**Comment by spwoodcock** [comment]
>  API requirements: 
> 1. Org admin list of specific org
> 2. Project summary list related to org (with pagination & search)
> 
> CC: @Sujanadh @spwoodcock 

1. Could be a separate endpoint, or worked into an existing one as a param.

2. Should be a param on the existing project endpoint probably.

**Comment by NSUWAL123** [comment]
Also, one thing that need to be included in the project summaries API according to the new design: 
1. Mapped, validated count
2. Total submission count
![Image](https://github.com/user-attachments/assets/944e9c97-3534-4e98-bcb0-e93d1b3c4c96)

@Sujanadh

**Comment by manjitapandey** [comment]
Organisation should have at least one organisation admin. 
An admin can't remove themselves from their role. Basically, no user can update their role by themselves as this causes clash on access and I don't get why would someone change their roles by their own. 

These things can be changed.

**Comment by manjitapandey** [comment]
Org should not be deleted if it contains any active projects.

**Comment by Sujanadh** [comment]
Organization is at the top of the hierarchy. Why would they want to delete the organization and keep the projects from those organizations? Moreover, we have `organisation_id` as a foreign key constraint on the project. We can't delete an organization unless the projects within the organization are deleted. If some projects do not have ODK credentials set, they depend on the organization, so deleting the organization would also bring issues to the project.

**Comment by spwoodcock** [comment]
> Organization is at the top of the hierarchy. Why would they want to delete the organization and keep the projects from those organizations? Moreover, we have `organisation_id` as a foreign key constraint on the project. We can't delete an organization unless the projects within the organization are deleted. If some projects do not have ODK credentials set, they depend on the organization, so deleting the organization would also bring issues to the project.

100% agree - we never want a scenario when an org is deleted and projects still exist.

By having the code as:

```sql
WITH deleted_org_managers AS (
    DELETE FROM organisation_managers
    WHERE organisation_id = %(org_id)s
    RETURNING organisation_id
), deleted_org AS (
    DELETE FROM organisations
    WHERE id = %(org_id)s
    RETURNING id
)
SELECT id FROM deleted_org;
```

this would allow users to delete organisations with no projects in them.
Say the user just created an org, but decide they actually don't want it (🤷).

But in most cases, when there are actually projects in the org, the deletion would fail (foreign key constraints as you say).

The current setup allow for the deletion of an entire organisation, perhaps accidentally by the admin or org manager.

We have two options:
1. Keep the code as it is, but never expose this endpoint as an option in the UI. Only admins should be able to do this, via API call (or directly in the db of course).
2. Update the code to be simpler, as above, giving the option in the UI, but returning an error like: `Your organisation contains projects, so deletion is not possible via this method. Please contact the instance admin'.

**Comment by manjitapandey** [comment]
I agree with second option, we can return error msg in a pop-up as: `This organization cannot be deleted as it contains existing projects. Please delete all associated projects before attempting to delete the organization.`

**Comment by Sujanadh** [comment]
> > Organization is at the top of the hierarchy. Why would they want to delete the organization and keep the projects from those organizations? Moreover, we have `organisation_id` as a foreign key constraint on the project. We can't delete an organization unless the projects within the organization are deleted. If some projects do not have ODK credentials set, they depend on the organization, so deleting the organization would also bring issues to the project.
> 
> 100% agree - we never want a scenario when an org is deleted and projects still exist.
> 
> By having the code as:
> 
> WITH deleted_org_managers AS (
>     DELETE FROM organisation_managers
>     WHERE organisation_id = %(org_id)s
>     RETURNING organisation_id
> ), deleted_org AS (
>     DELETE FROM organisations
>     WHERE id = %(org_id)s
>     RETURNING id
> )
> SELECT id FROM deleted_org;
> this would allow users to delete organisations with no projects in them. Say the user just created an org, but decide they actually don't want it (🤷).
> 
> But in most cases, when there are actually projects in the org, the deletion would fail (foreign key constraints as you say).
> 
> The current setup allow for the deletion of an entire organisation, perhaps accidentally by the admin or org manager.
> 
> We have two options:
> 
> 1. Keep the code as it is, but never expose this endpoint as an option in the UI. Only **admins** should be able to do this, via API call (or directly in the db of course).
> 2. Update the code to be simpler, as above, giving the option in the UI for **org managers**, but returning an error like: `Your organisation contains projects, so deletion is not possible via this method. Please contact the instance admin'.

I think Option 2 would be better. Hiding the delete option might confuse users as to why there is no deletion option, and seeing the delete option in some organisations and not in others might create confusion.
